src/almanack/entropy.py

```diff
@@ -0,0 +1,39 @@
+"""
+This module finds the amount of entropy on a file based level
+"""
+
+import math
+
+from .git_parser import calculate_loc_changes
+
+
+def calculate_shannon_entropy(
+    repo_path: str, source_commit: str, target_commit: str, file_names: list[str]
+) -> dict[str, float]:
+    """
+    Calculates the Shannon entropy for changes in specified files between two commits.
+
+    Args:
+        repo_path (str): The path to the git repository.
+        source_commit (str): The hash of the source commit.
+        target_commit (str): The hash of the target commit.
+        file_names (list[str]): List of file names to calculate entropy for.
+
+    Returns:
+        dict[str, float]: A dictionary mapping file names to their calculated entropy.
+
+    """
+    changes = calculate_loc_changes(repo_path, source_commit, target_commit, file_names)
+    total_changes = sum(changes.values())
+
+    entropy = {}
```

Comment:
Consider using a more descriptive name for this variable to address the question: "what type of entropy?"